source/common/upstream/upstream_impl.cc

```diff
@@ -1199,11 +1166,22 @@ ClusterInfoImpl::ClusterInfoImpl(
                                "HttpProtocolOptions can be specified");
   }
 
-  // If load_balancing_policy is set we will use it directly, ignoring lb_policy.
+  // Handle the original_dst_lb_config case first.
+  if (config.lb_config_case() == envoy::config::cluster::v3::Cluster::kOriginalDstLbConfig) {
```

Comment:
this looks new.  is it?

Reply:
no, it's old config. But it's config for cluster provided load balancer and we cannot convert it as an extension config.

I want to move it to the original dst cluster directly in a new PR. (not sure if it works, but I will try).